KGP.TicketApp.Backend/Controllers/UsersController.cs

```diff
@@ -1,52 +1,116 @@
+        }
+        #endregion
+
         #region Post methods
 
         // POST users/clients/login
+        [AllowAnonymous]
         [HttpPost("clients/login")]
-        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(Client))]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(ClientDTO))]
         [ProducesResponseType(StatusCodes.Status400BadRequest)]
         public IActionResult PostClientsLogin([FromBody] LoginCredentialsRequest request)
         {
-            return BadRequest();
+            var user = repositoryWrapper.ClientRepository.FindUserByEmail(request.Email);
+            if (user == null)
+                return BadRequest("Email doesn't exist in the system");
+            if (!BCrypt.Net.BCrypt.Verify(request.Password, user.Password))
+                return BadRequest("Password is incorrect");
+
+            Response.Headers.Add("Token", JwtTokenHelper.CreateToken(request.Email, user.Id.ToString(), settings));
+
+            return Ok(new ClientDTO() { Id = user.Id.ToString(), Name = user.Name, Surname = user.Surname });
         }
 
-        // POST users/organizers/login        
+        // POST users/organizers/login
+        [AllowAnonymous]
         [HttpPost("organizers/login")]
-        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(Organizer))]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(OrganizerDTO))]
         [ProducesResponseType(StatusCodes.Status400BadRequest)]
         public IActionResult PostOrganizersLogin([FromBody] LoginCredentialsRequest request)
         {
-            return BadRequest();
+            var user = repositoryWrapper.OrganizerRepository.FindUserByEmail(request.Email);
+            if (user == null)
+                return BadRequest("Email doesn't exist in the system");
+            if (!BCrypt.Net.BCrypt.Verify(request.Password, user.Password))
+                return BadRequest("Password is incorrect");
+
+            Response.Headers.Add("Token", JwtTokenHelper.CreateToken(request.Email, user.Id.ToString(), settings));
+
+            return Ok(new OrganizerDTO() { Id = user.Id.ToString(), Name = user.Name, Surname = user.Surname, CompanyName = user.CompanyName });
         }
 
-        // POST users/registerOrganizer       
+        // POST users/registerOrganizer
+        [AllowAnonymous]
         [HttpPost("registerOrganizer")]
         [ProducesResponseType(StatusCodes.Status200OK)]
         [ProducesResponseType(StatusCodes.Status400BadRequest)]
         public IActionResult PostRegisterOrganizer([FromBody] RegisterOrganizerRequest request)
         {
-            return BadRequest();
+            repositoryWrapper.OrganizerRepository.Create(new Organizer()
+            {
+                Email = request.Email,
+                Name = request.Name,
+                Surname = request.Surname,
+                Password = BCrypt.Net.BCrypt.HashPassword(request.Password),
```

Comment:
usługi związane z tym BCryptem wrzuciłbym do jakiegoś serwisu BCryptPasswordService, który implementuje jakieś IPasswordService. To się przewija w kilku miejscach

Reply:
poprawię